services/web/server/src/simcore_service_webserver/login/_invitations_service.py

```diff
@@ -41,9 +41,9 @@
 )
 from ..products.models import Product
 from ..users import users_service
-from . import _auth_service, _confirmation_service
-from ._login_repository_legacy import (
-    AsyncpgStorage,
+from . import _auth_service
+from ._controller.rest._rest_dependencies import get_confirmation_service
```

Comment:
Well, I would put an interface to hide this. You're in service layer here, and you're explicitly using an import to a particular (REST) implementation of a secondary adapter. If the technology changes you will  have to touch this module as well: coupling.